File: 3242-count-elements-with-maximum-frequency/count-elements-with-maximum-frequency.java
class Solution {
    public int maxFrequencyElements(int[] nums) {
        HashMap<Integer, Integer> mp = new HashMap<>();

        for (int it: nums) 
            mp.put(it, mp.getOrDefault(it, 0) + 1);
        
        int maxi = 0;
        for (int it: mp.values()) {
            maxi = Math.max(maxi, it);
        }

        int ans = 0;
        for (int it: mp.values())
        {
            if (it == maxi)
                ans += maxi;
        }

        return ans;
    }
}

File: 3242-count-elements-with-maximum-frequency/count-elements-with-maximum-frequency.cpp
class Solution {
public:
    int maxFrequencyElements(vector<int>& nums) {
        unordered_map<int, int> mpp;
        for (auto it: nums) {
            mpp[it]++;
        }

        int maximum_freq;
        for (auto it: mpp) 
        {
            maximum_freq = max( maximum_freq, it.second);
        }

        int maximum_freq_count = 0;
        for (auto it: mpp) {
            if (it.second == maximum_freq)
                maximum_freq_count += maximum_freq;
        }

        return maximum_freq_count;
    }
};